lib/core/src/Cardano/Wallet/DB/Sqlite/TH.hs

```diff
@@ -253,18 +256,53 @@ SeqStateAddress
     seqStateAddressSlot             SlotNo             sql=slot
     seqStateAddressAddress          W.Address          sql=address
     seqStateAddressIndex            Word32             sql=address_ix
-    seqStateAddressAccountingStyle  W.AccountingStyle  sql=accounting_style
+    seqStateAddressRole             W.Role             sql=role
     seqStateAddressStatus           W.AddressState     sql=status
 
     Primary
         seqStateAddressWalletId
         seqStateAddressSlot
         seqStateAddressAddress
         seqStateAddressIndex
-        seqStateAddressAccountingStyle
+        seqStateAddressRole
     Foreign Checkpoint seq_state_address seqStateAddressWalletId seqStateAddressSlot ! ON DELETE CASCADE
     deriving Show Generic
 
+-- Mapping of pool verification key hashes to indices, and the slot
+-- when they were discovered.
+SeqStateKeyHash
+    seqStateKeyHashWalletId         W.WalletId         sql=wallet_id
+    seqStateKeyHashSlot             SlotNo             sql=slot
+    seqStateKeyHashKeyHash          KeyHash            sql=verification_key_hash
+    seqStateKeyHashIndex            Word32             sql=verification_key_hash_ix
+    seqStateKeyHashStatus           W.AddressState     sql=status
+
+    Primary
+        seqStateKeyHashWalletId
+        seqStateKeyHashSlot
+        seqStateKeyHashKeyHash
+        seqStateKeyHashIndex
+    Foreign Checkpoint seq_state_key_hash seqStateKeyHashWalletId seqStateKeyHashSlot ! ON DELETE CASCADE
+    deriving Show Generic
+
+-- Mapping of discovered script hashes to our verification keys, and the slot
+-- when they were discovered.
+SeqStateScriptHash
+    seqStateScriptHashWalletId              W.WalletId         sql=wallet_id
+    seqStateScriptHashSlot                  SlotNo             sql=slot
+    seqStateScriptHashScriptHash            ScriptHash         sql=script_hash
+    seqStateScriptHashVerificationKeyIndex  Word32             sql=verification_key_ix
+    seqStateScriptHashKeysIndex             Word32             sql=keys_ix
```

Comment:
What's the difference between VerificationKeyIndex and KeysIndex here ?

Reply:
VerificationKeyIndex is the index of verification key. KeysIndex is index showing a position in list. We have scriptHash that can contain more than one of our keys, hence this index

Reply:
done in https://github.com/input-output-hk/cardano-wallet/pull/2308/commits/32ce104f76d3cf1efd573bc8dee37b3508334df5